src/log-analytics/azext_loganalytics/_help.py

```diff
@@ -7,22 +7,20 @@
 
 # pylint: disable=line-too-long
 
+helps['monitor log-analytics'] = """
+    type: group
+    short-summary: Commands for querying data in Log Analytics workspaces.
+"""
+
 helps['monitor log-analytics query'] = """
     type: command
     short-summary: Query a Log Analytics workspace.
     parameters:
-        - workspace: --workspace -w
-          type: string
-          short-summary: GUID of the Log Analytics workspace.
-        - analytics-query: --analytics-query
-          type: string
-          short-summary: Query to execute over the Log Analytics data.
-        - timespan: --timespan -t
-          type: string
-          short-summary: Timespan over which to query data. Defaults to all data.
-        - workspaces: --workspaces
-          type: array
-          short-summary: Additional workspaces to union data for querying. Specify additional workspace IDs separated by commas.
+      - name: --analytics-query
```

Comment:
Is there a particular reason we have this parameter's help here? If you aren't adding any other attributes to this parameter's help like `long-summary` then it's best to remove this parameter's help as well.

Reply:
Was this before or after adding the help information for this parameter in _params.py. This [line](https://github.com/Azure/azure-cli-extensions/blob/24f3fcd2c3aa81b8a157339fb998fc09c325fec7/src/log-analytics/azext_loganalytics/_params.py#L12) of code in _params.py should provide the help summary. If you add it, you shouldn't need to add this parameter's help information in _help.py. Just make sure the string you pass in for help is the short summary you need.

Reply:
Please try this out and let me know if it works.

Reply:
Using the following code (snippets): 

```python
def load_arguments(self, _):
    with self.argument_context('monitor log-analytics query') as c:
        c.argument('workspace', options_list=['--workspace', '-w'], help='GUID of the Log Analytics Workspace')
        c.argument('analytics-query', options_list=['--analytics-query'], help='Query to execute over Log Analytics data.')
        c.argument('timespan', options_list=['--timespan', '-t'], help='Timespan over which to query. Defaults to querying all available data.')
        c.argument('workspaces', nargs='+', help='Additional workspaces to union data for querying. Specify additional workspace IDs separated by space.')
```

```python
helps['monitor log-analytics query'] = """
    type: command
    short-summary: Query a Log Analytics workspace.
    examples:
      - name: Execute a simple query over past 3.5 days.
        text: |
          az monitor log-analytics query -w b8317023-66e4-4edc-8a5b-7c002b22f92f --analytics-query "AzureActivity | summarize count() by bin(timestamp, 1h)" -t P3DT12H
"""
```

I see the following results (note the missing text on the first line):

![image](https://user-images.githubusercontent.com/6800857/46115115-81bc4300-c1aa-11e8-9d1d-245d348035fb.png)

Reply:
Did you rebuild the whl file and re-add your extension after making these changes?

Reply:
Yes, I rebuilt the wheel and added the extension. This is not from a previous version of code. I ran this right now. The commands are cut off in the screenshot, this is the exact sequence I ran.

```powershell
rmdir -Force ./dist
python setup.py bdist_wheel
az extension remove -n log-analytics
az extension add -s ./dist/log_analytics-0.1.4-py2.py3-none-any.whl
az monitor log-analytics query -h
```

Reply:
(to be clear, I haven't changed the version again locally which is why it matches the one in PyPi)

Reply:
Okay, this is strange, let me look into this. I will checkout your pr.

Reply:
@alexeldeib after a while of debugging. I found the issue. 

TL;DR

your destination command parameter, **analytics-query**, name in _params.py does not match your method parameter name, **analytics_query**, in custom.py.

P.s. you do not need `options_list` if you parameter name is the same as your method name with the hyphens swapped for underscores and without the first two hyphens. 

See [here](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#customize-arguments) for info on customizing command arguments.

======================================================

Long explanation:

so if you look in _params.py you have:

```
...
        c.argument('analytics-query', options_list=['--analytics-query'], help='Query to execute over Log Analytics data.')
                          ^
                          |
dest method parameter name

...
```
whereas in custom.py you have:

```
...
def execute_query(client, workspace, analytics_query, timespan=None, workspaces=None):
                                         ^
                                         |
               method parameter name
...
``` 
Do you see the difference between the variable your are registering the parameter with and the variable you are passing to your method?

When customizing your parameter, you are saying `--analytics-query` maps to some method parameter `analytics-query`, that you want to add help text to. However this function parameter does not exist in your method definition. 

On the other hand you did not explicitly customize the `analytics_query` method param. So it has no special help or options. The cli automatically generates a vanilla command parameter of `--analytics-query`. 

By the way if your destination method parameter name is correct, you do not need to add an option list. The cli would know to convert your method parameter names with underscores to a suitable command parameter name with hyphens.

Reply:
Totally missed the hyphen vs. underscore, thank you for helping with the debug! And my apologies for sending you on a lengthy debug. I can make the code change and bump version, or leave as is? Alternatively, I can move storage from PyPi to an Azure blob and stick with 0.1.3 version as it *should* be.

Reply:
Oh no worries. I missed it too, lol. If you can move your storage to an azure blob, that would be nice. Would the whl file being on pypi affect users of your extension? Would you be able to remove or update the whl file on pypi so as not to confuse your users?

Reply:
After I upload to an Azure blob, I'll delete the PyPi project. Should have zero impact.